update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,341 @@
+
+        var responseEntity = response.getBodyObject(ApiUpdateDoiResponse.class);
+        assertThat(responseEntity, is(equalTo(ApiUpdateDoiResponse.newBuilder()
+            .withTask(ApiTask.newBuilder()
+                .withIdentifier(publication.getIdentifier().toString())
+                .withHref("https://mocked-hostname.example.net/publication/" + publication.getIdentifier().toString())
+                .build())
+            .build())));
+    }
+
+    private String validUsername(Publication publication) {
+        return publication.getOwner();
+    }
+
+    private DoiRequestSummary readPublicationDirectlyFromDynamo(UUID id) throws NotFoundException {
+        return doiRequestsService.fetchDoiRequest(id).orElseThrow();
+    }
+
+    private DoiRequestSummary expectedDoiRequestSummary(Publication publication) {
+        var includedDoiRequest = new DoiRequest.Builder()
+                .withDate(mockNow)
+                .withStatus(DoiRequestStatus.APPROVED)
+                .build();
+        return new DoiRequestSummary(
+                publication.getIdentifier(),
+                publication.getOwner(),
+                includedDoiRequest,
+                publication.getEntityDescription());
+    }
+
+    private Publication insertPublicationWithDoiRequest(Clock clock)
+        throws com.fasterxml.jackson.core.JsonProcessingException {
+        Publication publication = PublicationGenerator.getPublicationWithDoiRequest(clock)
+            .copy()
+            .withCreatedDate(mockOneHourBefore)
+            .build();
+        insertPublication(publicationsTableName, publication);
+        return publication;
+    }
+
+    private Publication insertPublicationWithoutDoiRequest(Clock clock)
+        throws com.fasterxml.jackson.core.JsonProcessingException {
+        Publication publication = PublicationGenerator.getPublicationWithoutDoiRequest(clock)
+                .copy()
+                .withCreatedDate(mockOneHourBefore)
+                .build();
+        insertPublication(publicationsTableName, publication);
+        return publication;
+    }
+
+    private ApiUpdateDoiRequest requestWithoutPublicationId() {
+        return new ApiUpdateDoiRequest();
+    }
+
+    private <T> GatewayResponse<T> sendRequest(ApiUpdateDoiRequest doiRequest, String username) throws IOException {
+        var pathParams = Map.of("publicationIdentifier", doiRequest.getPublicationId());
+        InputStream input = createRequest(doiRequest, username, pathParams);
+        ByteArrayOutputStream output = outpuStream();
+        handler.handleRequest(input, output, context);
+
+        return GatewayResponse.fromOutputStream(output);
+    }
+
+    private <T> GatewayResponse<T> sendRequest(ApiUpdateDoiRequest doiRequest, JsonNode requestContext)
+        throws IOException {
+        var pathParams = Map.of("publicationIdentifier", doiRequest.getPublicationId());
```

Comment:
magic string

Reply:
We should probably discuss this later if everything needs to be constants or not..

Reply:
Maybe, but this is a path parameter that you are expecting. You are formatting the input as the handler expects it. So it should refer to the class that expects this string. 

In general we are trying to avoid arbitrary strings in the code. A constant, except for the ability it gives you to not copy-paste, it also creates a context. If I want to understand why you are setting this parameter there, I should do a search in the whole project and try to extrapolate the role of this parameter. But if it were a constant with the proper name, then I would understand immediately.

Reply:
Fixed into a constant.